### PR TITLE
fix failing cypress tests

### PR DESCRIPTION
**Pull request by lubega-deriv**
Updated locators for the following tests:
- [x] cryptoWithdrawal
- [x] demoResetBalance
- [x] demoTransactions
- [x] fiatTransactions
- [x] fiatWithdrawal

### CODE REVIEW

File: cypress/e2e/wipWallet/demoResetBalance.cy.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-  cy.get('[class*="virtual"].wallets-accordion__header--virtual')
-    .find('.wallets-accordion__dropdown > svg')
-    .click()
-  cy.get('[class*="virtual"].wallets-accordion__header--virtual')
-    .contains('Reset balance')
-    .click()
+  cy.get('.wallets-dropdown__button').click()
+  cy.findByText('USD Demo Wallet').scrollIntoView()
+  cy.get('.wallets-list-card-dropdown__item-content').contains('USD Demo Wallet').click()
+  cy.get('.wallets-list-details__content').within(()=>{
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoResetBalance.cy.js:7]: do we not have any Testing library locators for this?

File: cypress/e2e/wipWallet/demoResetBalance.cy.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+    cy.contains('USD').should('be.visible')
+  })
+  cy.contains('Reset balance').should('be.visible')
+  cy.contains('Reset balance').click()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoResetBalance.cy.js:11]: can you please use Testing library locators here 
```suggestion
    cy.findByText('USD').should('be.visible')
  })
  cy.findByText('Reset balance').should('be.visible')
  cy.findByText('Reset balance').click()
```

File: cypress/e2e/wipWallet/demoTransactions.cy.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-    .contains('Reset balance')
-    .click()
+  cy.get('.wallets-dropdown__button').click()
+  cy.findByText('USD Demo Wallet').scrollIntoView()
+  cy.get('.wallets-list-card-dropdown__item-content').contains('USD Demo Wallet').click()
+  cy.get('.wallets-list-details__content').within(()=>{
+    cy.contains('USD').should('be.visible')
+  })
+  cy.contains('Reset balance').should('be.visible')
+  cy.contains('Reset balance').click()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:11]: same as above

File: cypress/e2e/wipWallet/demoTransactions.cy.js [hunk 4]
```diff
@@ -22,6 +23,24 @@ function reset_balance_demo() {
...
+    .type('1.000')
+  cy.get('form')
+    .findByRole('button', { name: 'Transfer', exact: true })
+    .should('be.enabled')
+    .click()
+  cy.findByText('Your transfer is successful!', {
+    exact: true,
+  }).should('be.visible')
+  cy.findByRole('button', { name: 'Make a new transfer' }).click()
+  cy.contains(/Transfer from/)
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:41]: same as above

File: cypress/e2e/wipWallet/demoTransactions.cy.js [hunk 5]
```diff
@@ -38,33 +57,27 @@ describe('WALL-2760 - Transfer and check transactions for Demo wallet', () => {
...
     cy.findByRole('button', { name: 'Transactions' }).click()
-    cy.get('#downshift-0-toggle-button').findByRole('button').click()
+    cy.findByTestId('dt_wallets_textfield_icon_right').findByRole('button').click()
     cy.findByRole('option', { name: 'Reset balance' }).click()
     cy.contains('+10,000.00 USD')
-    cy.get('[class=wallets-textfield__box').click()
+    cy.findByTestId('dt_wallets_textfield_icon_right').findByRole('button').click()
     cy.findByRole('option', { name: 'Transfer' }).click()
+    cy.contains('MT5 Derived')
+    cy.contains('Deriv X')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:80]: same as above

File: cypress/e2e/wipWallet/demoTransactions.cy.js [hunk 6]
```diff
@@ -38,33 +57,27 @@ describe('WALL-2760 - Transfer and check transactions for Demo wallet', () => {
...
-      .eq(1)
-      .click()
-      .type('1.000')
-    cy.get('form')
-      .findByRole('button', { name: 'Transfer', exact: true })
-      .should('be.enabled')
-      .click()
-    cy.findByText('Your transfer is successful!', {
+    demo_transfer(/MT5 Derived/)
+    cy.contains(/Transfer from/).click()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:61]: same as above

File: cypressTemplate.config.js
Comment (bilalmajid-deriv) [inline cypressTemplate.config.js:1]: can you remove this file it is no longer needed

File: cypress/e2e/wipWallet/fiatWithdrawal.cy.js [hunk 7]
```diff
@@ -22,27 +22,22 @@ describe('WALL-2830 - Fiat withdrawal send email', () => {
...
-    cy.then(() => {
-      cy.c_visitResponsive(
-        `${withdrawal_url}?verification=${verification_code}`,
-        'large'
-      )
+    beforeEach(() => {
+      cy.c_login('doughflow')
+      cy.c_visitResponsive('/wallets', 'large')
+      cy.contains('Wallet', { timeout: 10000 }).should('exist')
+      cy.contains('Withdraw').click()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/fiatWithdrawal.cy.js:29]: same as above

**Review by bilalmajid-deriv — CHANGES_REQUESTED** [review]

**Comment by bilalmajid-deriv** [comment]
@lubega-deriv can you please run `npm install` before committing as well
so the check for code formatting passes thanks

File: cypress/e2e/wipWallet/demoResetBalance.cy.js (hunk 1 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoResetBalance.cy.js]: In this situation no as it's only a div with a classname prop and no other locators
image

File: cypress/e2e/wipWallet/demoResetBalance.cy.js (hunk 2 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoResetBalance.cy.js]: Updated

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 3 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoTransactions.cy.js]: Updated

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 4 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoTransactions.cy.js]: Updated

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 5 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoTransactions.cy.js]: Updated

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 6 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoTransactions.cy.js]: Updated

File: cypressTemplate.config.js
Comment (lubega-deriv) [reply cypressTemplate.config.js:1]: Updated

File: cypress/e2e/wipWallet/fiatWithdrawal.cy.js (hunk 7 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/fiatWithdrawal.cy.js]: Updated

File: cypress/e2e/wip/Wallets/cryptoDeposit.cy.js [hunk 8]
```diff
@@ -38,12 +38,12 @@ describe('WALL-2831 - Crypto deposit and fiat onramp', () => {
     cy.get('.wallets-list-details__content').within(() => {
       cy.contains('BTC').should('be.visible')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/Wallets/cryptoDeposit.cy.js:39]:
```suggestion
      cy.findByText('BTC').should('be.visible')
```

File: cypress/e2e/wip/Wallets/cryptoDeposit.cy.js [hunk 9]
```diff
@@ -17,15 +17,15 @@ describe('WALL-2831 - Crypto deposit and fiat onramp', () => {
...
-    cy.contains('Deposit').click()
+    cy.findByText('Deposit').should('be.visible')
+    cy.findByText('Deposit').click()
     cy.get('canvas').should('be.visible')
-    cy.contains('Transaction status')
-    cy.contains(/To avoid loss of funds/)
+    cy.findByText('Transaction status')
+    cy.findByText(/To avoid loss of funds/)
     cy.get('.wallets-clipboard').click()
     cy.contains('Copied')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/Wallets/cryptoDeposit.cy.js:26]:
```suggestion
    cy.findByText('Copied')
```

File: cypress/e2e/wip/Wallets/cryptoTransactions.cy.js [hunk 10]
```diff
@@ -54,13 +54,13 @@ describe('WALL-2858 - Crypto transfer and transactions', () => {
...
     cy.findByRole('option', { name: 'Deposit' }).click()
-    cy.contains('+5.00000000 BTC')
+    cy.findByText('+5.00000000 BTC')
     cy.findByTestId('dt_wallets_textfield_box').click()
     cy.findByRole('option', { name: 'Withdrawal' }).click()
-    cy.contains('No recent transactions')
+    cy.findByText('No recent transactions')
     cy.findByTestId('dt_wallets_textfield_box').click()
     cy.findByRole('option', { name: 'Transfer' }).click()
     cy.contains('LTC Wallet')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wip/Wallets/cryptoTransactions.cy.js:66]:
```suggestion
    cy.findByText('LTC Wallet')
```

File: cypress/e2e/wipWallet/demoResetBalance.cy.js [hunk 11]
```diff
@@ -1,21 +1,24 @@
...
-    .click()
-  cy.get('[class*="virtual"].wallets-accordion__header--virtual')
-    .contains('Reset balance')
+  cy.get('.wallets-dropdown__button').click()
+  cy.findByText('USD Demo Wallet').scrollIntoView()
+  cy.get('.wallets-list-card-dropdown__item-content')
+    .contains('USD Demo Wallet')
     .click()
+  cy.get('.wallets-list-details__content').within(() => {
+    cy.contains('USD').should('be.visible')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoResetBalance.cy.js:10]:
```suggestion
    cy.findByText('USD').should('be.visible')
```

File: cypress/e2e/wipWallet/demoResetBalance.cy.js [hunk 12]
```diff
@@ -1,21 +1,24 @@
...
+  })
+  cy.findByText('Reset balance').should('be.visible')
+  cy.findByText('Reset balance').click()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoResetBalance.cy.js:13]:
```suggestion
  cy.findByText('Reset balance').should('be.visible').click()
```

File: cypress/e2e/wipWallet/demoTransactions.cy.js [hunk 13]
```diff
@@ -1,27 +1,48 @@
...
-    .click()
-  cy.get('[class*="virtual"].wallets-accordion__header--virtual')
-    .contains('Reset balance')
+  cy.get('.wallets-dropdown__button').click()
+  cy.findByText('USD Demo Wallet').scrollIntoView()
+  cy.get('.wallets-list-card-dropdown__item-content')
+    .contains('USD Demo Wallet')
     .click()
+  cy.get('.wallets-list-details__content').within(() => {
+    cy.contains('USD').should('be.visible')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:10]:
```suggestion
    cy.findByText('USD').should('be.visible')
```

File: cypress/e2e/wipWallet/demoTransactions.cy.js [hunk 14]
```diff
@@ -1,27 +1,48 @@
...
+  })
+  cy.findByText('Reset balance').should('be.visible')
+  cy.findByText('Reset balance').click()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:13]:
```suggestion
  cy.findByText('Reset balance').should('be.visible').click()
```

File: cypress/e2e/wipWallet/demoTransactions.cy.js [hunk 15]
```diff
@@ -33,38 +54,36 @@ describe('WALL-2760 - Transfer and check transactions for Demo wallet', () => {
...
-    cy.contains('+10,000.00 USD')
-    cy.get('[class=wallets-textfield__box').click()
+    cy.findByText('+10,000.00 USD')
+    cy.findByTestId('dt_wallets_textfield_icon_right')
+      .findByRole('button')
+      .click()
     cy.findByRole('option', { name: 'Transfer' }).click()
     cy.contains('-10.00 USD')
+    cy.contains('MT5 Derived')
+    cy.contains('Deriv X')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:87]:
```suggestion
    cy.findByText('MT5 Derived')
    cy.findByText('Deriv X')
```

File: cypress/e2e/wipWallet/fiatTransactions.cy.js [hunk 16]
```diff
@@ -45,14 +45,20 @@ describe('WALL-2858 - Fiat transfer and transactions', () => {
...
     cy.findByRole('option', { name: 'Withdrawal' }).click()
-    cy.contains('No recent transactions')
-    cy.get('#downshift-0-toggle-button').findByRole('button').click()
+    cy.findByText('No recent transactions')
+    cy.findByTestId('dt_wallets_textfield_icon_right')
+      .findByRole('button')
+      .click()
     cy.findByRole('option', { name: 'Transfer' }).click()
     cy.contains('LTC Wallet')
     cy.contains('ETH Wallet')
```
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/fiatTransactions.cy.js:64]:
```suggestion
    cy.findByText('LTC Wallet')
    cy.findByText('ETH Wallet')
```

**Review by bilalmajid-deriv — CHANGES_REQUESTED** [review]
Changes recommended to update to use Testing Library locators thanks.

File: cypress/e2e/wip/Wallets/cryptoDeposit.cy.js (hunk 8 above)
Comment (lubega-deriv) [reply cypress/e2e/wip/Wallets/cryptoDeposit.cy.js:39]: Invalid locator and unable to find element which matches

File: cypress/e2e/wip/Wallets/cryptoDeposit.cy.js (hunk 9 above)
Comment (lubega-deriv) [reply cypress/e2e/wip/Wallets/cryptoDeposit.cy.js:26]: Invalid locator and unable to find element which matches

File: cypress/e2e/wip/Wallets/cryptoTransactions.cy.js (hunk 10 above)
Comment (lubega-deriv) [reply cypress/e2e/wip/Wallets/cryptoTransactions.cy.js:66]: Invalid locator and unable to find element which matches

File: cypress/e2e/wipWallet/demoResetBalance.cy.js (hunk 11 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoResetBalance.cy.js:10]: Invalid locator and unable to find element which matches

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 13 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoTransactions.cy.js:10]: Invalid locator and unable to find element which matches

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 15 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/demoTransactions.cy.js:87]: Invalid locator and unable to find element which matches

File: cypress/e2e/wipWallet/fiatTransactions.cy.js (hunk 16 above)
Comment (lubega-deriv) [reply cypress/e2e/wipWallet/fiatTransactions.cy.js:64]: Invalid locator and unable to find element which matches

File: cypress/e2e/wip/Wallets/cryptoDeposit.cy.js (hunk 8 above)
Comment (bilalmajid-deriv) [reply cypress/e2e/wip/Wallets/cryptoDeposit.cy.js:39]: edited previous suggestion can try that now

File: cypress/e2e/wip/Wallets/cryptoDeposit.cy.js (hunk 9 above)
Comment (bilalmajid-deriv) [reply cypress/e2e/wip/Wallets/cryptoDeposit.cy.js:26]: edited previous suggestion can try that now

File: cypress/e2e/wip/Wallets/cryptoTransactions.cy.js (hunk 10 above)
Comment (bilalmajid-deriv) [reply cypress/e2e/wip/Wallets/cryptoTransactions.cy.js:66]: edited previous suggestion can try that now

File: cypress/e2e/wipWallet/demoResetBalance.cy.js (hunk 11 above)
Comment (bilalmajid-deriv) [reply cypress/e2e/wipWallet/demoResetBalance.cy.js:10]: edited previous suggestion can try that now

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 13 above)
Comment (bilalmajid-deriv) [reply cypress/e2e/wipWallet/demoTransactions.cy.js:10]: edited previous suggestion can try that now

File: cypress/e2e/wipWallet/demoTransactions.cy.js (hunk 15 above)
Comment (bilalmajid-deriv) [inline cypress/e2e/wipWallet/demoTransactions.cy.js:86]:
```suggestion
    cy.findByText(/-10.00 USD/)
    cy.findByText(/MT5 Derived/)
    cy.findByText(/Deriv X/)
```

File: cypress/e2e/wipWallet/fiatTransactions.cy.js (hunk 16 above)
Comment (bilalmajid-deriv) [reply cypress/e2e/wipWallet/fiatTransactions.cy.js:64]: edited previous suggestion can use that

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Pull request merged**